### ISSUE TITLE
Elements white covered in Ionic 3 using Chrome 66

### ISSUE THREAD

**Issue by sagashiftyblow** [usablica/intro.js#840]
## Description
Element to be highlighted is covered (instead of highlighted) by the white underlying intro js overlay in a Ionic 3 app. It happens only on chrome. Works fine on Firefox and Explorer. I found some other reports of people trying to use introjs with ionic failing for this reason.
![example](https://user-images.githubusercontent.com/17250898/38958226-f246ebc4-435c-11e8-84aa-18a344055870.PNG)
. 

## Example (recommended)
The issue is showing [here](https://stackblitz.com/edit/ionic-5d3qcz?file=pages%2Fhome%2Fhome.html).

## Environment (optional)
Tried with the introjs npm version, importing in index.html version 2.7. The process used to import introjs in Ionic is described [here](https://stackoverflow.com/questions/46780985/intro-js-with-ionic-3-angular) but it's a bit chaotic. I made a summary [here](https://stackoverflow.com/questions/48201970/how-to-implement-walk-through-or-tour-in-ionic-3). The issue showed up in Chrome 66, with Ionic 3.

**Comment by jshupptechnologies** [comment]
@sagashiftyblow  Any solution on this ? I am struggling with same issue.

**Comment by sagashiftyblow** [comment]
Unfortunately, no... @jshupptechnologies

**Comment by sagashiftyblow** [comment]
If you find any solution please share @jshupptechnologies Thank you!!

**Comment by jwbeech** [comment]
Trying to run down the change that caused this issue as I need to use this library too. 
@sagashiftyblow did you confirm this is working in chrome 65 as in something that was in 66 caused the issue?
Or could it be in an earlier build of chrome too?

**Comment by sagashiftyblow** [comment]
Truly, I don't know. I found the issue on Chrome 66, and it's still here on Chrome 67. I haven't tried on earlier versions, unfortunately. Seems, trivially, to be a problem linked to the management of z-index, but I couldn't found a solution still. If I can help, feel free to ask.

**Comment by jwbeech** [comment]
@sagashiftyblow after much trial and error I found the problem / solution.
The issue is a css property attached to the root ion-app element: `contain: strict;`.
This appears to be a performance tweak for the ionic app which clearly has z-index issues in chrome. 

`contain: strict;` is actually shorthand for `contain: layout paint size style`, changing it to: `contain: layout size style;` is enough to mitigate the problem. 
Some background I found: https://termvader.github.io/css-contain/

The following SCSS should allow for this change only when the introjs tips are applied:

```
ion-app, ion-nav, ion-tab, ion-tabs, .app-root, .ion-page{
    &.introjs-fixParent{
        contain: layout size style;
    }
}
```

There may be other layouts that require this change not covered by the above style but that is the crux of the issue.

**Comment by sagashiftyblow** [comment]
It works!! Great! Thank you very very very much for sharing solution!

**Comment by jwbeech** [comment]
@afshinm considering the intro.js framework supports a work around for adding a class to address parent container issues: i.e. introjs-fixParent.
Don't you think this fix should be integrated into the framework at large. This problem is not restricted to ionic but to parent elements with `contain: strict` on chrome.

**Comment by Stiveknx** [comment]
I was having problems to highlight itens inside <ion-header> <ion-toolbar> on ionic, even after @jwbeech  solution.
To fix it, I just changed to all elements with .introjs-fixParent to have the contain: layout size style.

```
.introjs-fixParent {
	contain: layout size style !important;
}
```

**Comment by Stiveknx** [comment]
Have tou tried to add the CSS rule i've posted?

Em 22 de ago de 2018 19:21, "Martial Anouman" <notifications@github.com>
escreveu:

Hi, i got the same issue while trying to highlight an element inside an
ion-card. Any idea ?

—
You are receiving this because you commented.
Reply to this email directly, view it on GitHub
<https://github.com/usablica/intro.js/issues/840#issuecomment-415200894>,
or mute the thread

**Comment by OAnouman** [comment]
Yes i did @Stiveknx. It works on all components i used so far but on ion-card. I don't know why. I tried to dive in the CSS rules but nothing worked for now.

**Comment by Stiveknx** [comment]
Try to add position: absolute !important on introJs-fixParent.

Em qua, 22 de ago de 2018 19:52, Martial Anouman <notifications@github.com>
escreveu:

**Comment by OAnouman** [comment]
I found a fix. It was due to a css rule of .item class. contain was set to "content". Just set it to none or layout size style.
`.item {
 contain: layout size style | none !important;
}`

**Comment by OAnouman** [comment]
In general we got to deal with contain: strict spread all over ionic native components.
So if the highlighted item doesn't appear look for contain: strict in its parents.
Hope it will help someone.

**Comment by sagashiftyblow** [comment]
The same problem re-appear if I add a (transform: scale(1.5) !important;) to ion-content (dont' ask why, but I need to scale all the pages). The overlay is correctly positioned but appears with a white cover on it. Someone had this problem, too, maybe?

**Comment by DiegoFSFlor** [comment]
yes, I had the same problem, but I solved downloading the css into the project and adding this command inside the item.

Now I have another mistake, after the tour finishes, the button that was part of the tour does not let me click on it, has anyone else been through this?
While the tour is inside the button it works normally.

**Comment by fidaktk** [comment]
I had the same issue. adding box-shadow and making the overly transparent solved the issue.
`.introjs-helperLayer {
 background: transparent !important;
 box-shadow: 0 0 0 9999px rgba(1, 1, 1, 0.8);
}`

and 

`.introjs-overlay {
 background: transparent !important;
}`

**Comment by lsantaniello** [comment]
> ```
> ion-app, ion-nav, ion-tab, ion-tabs, .app-root, .ion-page{
>     &.introjs-fixParent{
>         contain: layout size style;
>     }
> }
> ```

Not works for me :( Where I have to add this code block?

**Comment by lsantaniello** [comment]
Solved appling this css code:

.introjs-helperLayer { background: transparent !important; box-shadow: 0 0 0 9999px rgba(1, 1, 1, 0.8); }
.introjs-overlay { background: transparent !important; }

It works for the content but I have problem for the header bar.

[https://ibb.co/17nYmww](this is the issue)

Could you please help me?

**Comment by lsantaniello** [comment]
> I was having problems to highlight itens inside `<ion-header> <ion-toolbar>` on ionic, even after @jwbeech solution.
> To fix it, I just changed to all elements with .introjs-fixParent to have the contain: layout size style.
> 
> ```
> .introjs-fixParent {
> 	contain: layout size style !important;
> }
> ```

Please @Stiveknx , help me. I have the same issue for ion-header component. How you solved it?

Thanks in advance

**Comment by yafp** [comment]
I had the same problem in an electron app (based on chrome) i am working on.
While introjs worked as expected for a while it suddenly just renderer white space above the highlighted ui element

Adding the following to the project css solved it

`.introjs-helperLayer { background: transparent !important; box-shadow: 0 0 0 9999px rgba(1, 1, 1, 0.8); }`

and

`.introjs-overlay { background: transparent !important; }`
